tensorflow/lite/micro/kernels/xcore/xcore_conv2d_v2.cc

```diff
@@ -0,0 +1,292 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "Conv2d.hpp"
+#include "tensorflow/lite/c/common.h"
+#include "tensorflow/lite/kernels/internal/tensor_ctypes.h"
+#include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/kernels/xcore/xcore_custom_options.h"
+#include "tensorflow/lite/micro/kernels/xcore/xcore_dispatcher.h"
+#include "tensorflow/lite/micro/kernels/xcore/xcore_op_utils.h"
+extern "C" {
+#include "nn_operator.h"
+}
+
+namespace tflite {
+namespace ops {
+namespace micro {
+namespace xcore {
+namespace conv_v2 {
+
+// -------------------------------------------------------------------- //
+// thread data type and worker functions
+// -------------------------------------------------------------------- //
+
+struct Conv2DThread {
+  int8_t* X;
+  int8_t* Y;
+  int8_t* scratch;
+  nn::Filter2D* f;
+};
+
+extern "C" {
+// TODO
+#pragma stackfunction 1000
```

Comment:
eek

Reply:
Yup, Henk helped me to get it working. Was not finding the stack size without it. We can look into it when you get back. :)

Reply:
I think the correct solution to this is to use CRTP as that will statically let the tools know what is going on. For now this is fine though.